api/oauth.go

```diff
@@ -6,3 +6,18 @@ type OAuthToken struct {
 	RefreshToken string `jsonapi:"attr,refreshToken"`
 	ExpiresTs    int64  `jsonapi:"attr,expiresAt"`
 }
+
+// OAuthConfig is the API message for OAuthConfig.
+type OAuthConfig struct {
+	Endpoint      string `jsonapi:"attr,endpoint"`
+	ApplicationID string `jsonapi:"attr,applicationId"`
+	Secret        string `jsonapi:"attr,secret"`
+	RedirectURL   string `jsonapi:"attr,redirectUrl"`
```

Comment:
nit: `redirectURL` is set to `$FRONTEND_HOST:$PORT/oauth/callback`, and we can directly derive this field in the backend. We may need not pass this field from the frontend.

Reply:
Good call, removed it completely.